test/Microsoft.Azure.Devices.Edge.Test/Context.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+
+    public class Context
+    {
+        static readonly Lazy<Context> Default = new Lazy<Context>(
+            () =>
+            {
+                IConfiguration context = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("context.json")
+                    .AddEnvironmentVariables("E2E_")
+                    .Build();
+
+                string GetString(string name) => context.GetValue<string>(name);
+
+                string GetStringOr(string name, string alternate)
+                {
+                    string value = GetString(name);
+                    return string.IsNullOrEmpty(value) ? alternate : value;
+                }
+
+                Option<(string, string, string)> AllOrNothing(string a, string b, string c) =>
+                    string.IsNullOrEmpty(a) || string.IsNullOrEmpty(b) || string.IsNullOrEmpty(c)
+                        ? Option.None<(string, string, string)>()
+                        : Option.Some((a, b, c));
+
+                return new Context
+                {
+                    ConnectionString = GetString("IOT_HUB_CONNECTION_STRING"),
+                    EventHubEndpoint = GetString("EVENT_HUB_ENDPOINT"),
+                    DeviceId = GetStringOr("deviceId", $"end-to-end-{Dns.GetHostName()}-{DateTime.Now:yyyy'-'MM'-'dd'T'HH'-'mm'-'ss'-'fff}"),
+                    InstallerPath = Option.Maybe(GetString("installerPath")),
+                    PackagePath = Option.Maybe(GetString("packagePath")),
+                    Proxy = Option.Maybe(context.GetValue<Uri>("proxy")),
+                    Registry = AllOrNothing(GetString("registry"), GetString("user"), GetString("CONTAINER_REGISTRY_PASSWORD")),
```

Comment:
Nit: This AllOrNothing might get confusing.. you might want to throw if one of the values is not present.

Reply:
True. I'll throw so that the caller will have to fix the bug in their test code.

Reply:
Fixed.